plugins/catalog-backend/src/ingestion/processors/GithubMultiOrgReaderProcessor.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  ScmIntegrations,
+  GithubCredentialsProvider,
+} from '@backstage/integration';
+import { buildOrgHierarchy } from './util/org';
+
+/**
+ * Extracts teams and users out of multiple GitHub orgs.
+ */
+export class GithubMultiOrgReaderProcessor implements CatalogProcessor {
+  private readonly integrations: ScmIntegrations;
+  private readonly logger: Logger;
+
+  static fromConfig(config: Config, options: { logger: Logger }) {
+    const integrations = ScmIntegrations.fromConfig(config);
+
+    return new GithubMultiOrgReaderProcessor({
+      ...options,
+      integrations,
+    });
+  }
+
+  constructor(options: { integrations: ScmIntegrations; logger: Logger }) {
+    this.integrations = options.integrations;
+    this.logger = options.logger;
+  }
+
+  async readLocation(
+    location: LocationSpec,
+    _optional: boolean,
+    emit: CatalogProcessorEmit,
+  ): Promise<boolean> {
+    if (location.type !== 'github-multi-org') {
+      return false;
+    }
+
+    const gitHubConfig = this.integrations.github.byUrl(location.target)
+      ?.config;
+    if (!gitHubConfig) {
+      throw new Error(
+        `There is no GitHub integration that matches ${location.target}. Please add a configuration entry for it under integrations.github`,
+      );
+    }
+
+    // TODO: grab installations for each provided app config if multiple are defined
+    const appClient = new Octokit({
+      baseUrl: gitHubConfig.apiBaseUrl,
+      headers: { Accept: 'application/vnd.github.machine-man-preview+json' },
+      authStrategy: createAppAuth,
+      auth: {
+        appId:
+          gitHubConfig.apps &&
+          gitHubConfig.apps[0] &&
+          gitHubConfig.apps[0].appId,
+        privateKey:
+          gitHubConfig.apps &&
+          gitHubConfig.apps[0] &&
+          gitHubConfig.apps[0].privateKey,
+      },
+    });
+
+    // TODO: support access token auth by grabbing all orgs accessible by the token  
+    const allOrgs = await getOrganizations(appClient);
+
+    const credentialsProvider = GithubCredentialsProvider.create(gitHubConfig);
+
+    const allUsersMap = new Map();
+    const baseUrl = new URL(location.target).origin;
+
+    for (const org of allOrgs) {
+      const { headers } = await credentialsProvider.getCredentials({
+        url: `${baseUrl}/${org}`,
+      });
+      const client = graphql.defaults({
+        baseUrl: gitHubConfig.apiBaseUrl,
+        headers,
+      });
+
+      const startTimestamp = Date.now();
+      this.logger.info(`Reading GitHub users and groups for org: ${org}`);
+      const { users } = await getOrganizationUsers(client, org);
+      const { groups, groupMemberUsers } = await getOrganizationTeams(
+        client,
+        org,
+        true,
+      );
+
+      const duration = ((Date.now() - startTimestamp) / 1000).toFixed(1);
+      this.logger.debug(
+        `Read ${users.length} GitHub users and ${groups.length} GitHub groups from ${org} in ${duration} seconds`,
```

Comment:
```suggestion
        `Read ${users.length} GitHub users and ${groups.length} GitHub teams from ${org} in ${duration} seconds`,
```

I know there's nuance here, in that GitHub calls them teams but Backstage groups, so we're essentially trying to ingest teams to MAKE them groups in Backstage, but considering this msg is specific about what it's done IN GitHub, maybe this should say teams?